### PR TITLE
[36] – Feature: Polling

### PR DESCRIPTION
**Pull request by Drish-xD**
Implemented polling strategy to ensure the data displayed in the UI is always up-to-date, enhancing the overall responsiveness and speed of the user interface.

Use Constant `POLLING_INTERVAL` to change the interval timing. 
Current its `30 sec`

Disabled edit if the session is in pending state.

### CODE REVIEW

File: src/Constants/index.ts [hunk 1]
```diff
@@ -3,9 +3,13 @@ import { DeepKeys } from '@tanstack/react-table';
 
 export * from './Options';
 
+// TABLE PAGINATION CONSTANTS
 export const DATA_PER_PAGE = 50;
 export const PAGE_SIZE_OPTIONS = [10, 25, 50, 100, 200];
 
+// TABLE POLLING CONSTANTS
```
Comment (suryabulusu) [inline src/Constants/index.ts:10]: we can improve this maybe. we are sure the lambda function takes ~30 sec to run... so maybe in the interval [20, 60] seconds we need to poll more frequently. can schedule it better

**Review by suryabulusu — COMMENTED** [review]
left a comment.. do write a description of PR

- do we also "gray" the row to indicate pending status ? or no need? up to you

File: src/types/api.types.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ const metaDataSchema = z.object({
   test_takers_count: z.number().int().optional(),
   test_type: z.string().optional(),
   subject: z.string().optional(),
+  status: z.enum(['success', 'pending', 'failed']).optional(),
```
Comment (suryabulusu) [inline src/types/api.types.ts:27]: for old quizzes.. status would be empty?

Comment (suryabulusu) [reply src/types/api.types.ts:27]: if session id is empty -- u can assume status is fail. if session id not empty.. u can assume success/?

**Comment by Drish-xD** [comment]
> left a comment.. do write a description of PR
> 
> * do we also "gray" the row to indicate pending status ? or no need? up to you

Faded the row is its status is pending.

File: src/types/api.types.ts (hunk 2 above)
Comment (Drish-xD) [reply src/types/api.types.ts]: I have handled the Status for old sessions also.
basically I have set it to success by default on UI.

File: src/hooks/usePolling.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    return { pendingSessions, stopPolling };
+  }, [data]);
+
+  useEffect(() => {
+    if (stopPolling) return;
+    const intervalId = setInterval(router.refresh, POLLING_INTERVAL);
+    return () => clearInterval(intervalId);
+  }, [stopPolling]);
+
+  console.info(`${new Date().toLocaleTimeString()} : `, {
```
Comment (suryabulusu) [inline src/hooks/usePolling.tsx:34]: remove before merge

**Review by suryabulusu — APPROVED** [review]

**Pull request merged**